src/xo-app/backup/restore/index.js

```diff
@@ -117,6 +158,19 @@ class _ModalBody extends Component {
         placeholder={intl.formatMessage(messages.importBackupModalSelectBackup)}
       />
       <br />
+      {vdis !== undefined && this.state.sr &&
+        <_Collapsible collapsible={vdis.length >= 3} buttonText={_('backupRestoreChooseSRForEachVdis')}>
+          <br />
+          {map(vdis, vdi =>
+            <span>
+              <b>{_('backupRestoreVdiLabel')}:</b> {vdi.name}
+              <SelectSr key={vdi.uuid} onChange={this.linkState(`mapVdisSrs.${vdi.uuid}`)} value={this._getSelectedValue(vdi.uuid)} predicate={this._srPredicate} />
```

Comment:
`this._getSelectedValue` is called in `render` and potentially triggers another render with `setState`. The `render` function should be [pure](https://facebook.github.io/react/docs/react-component.html#reference).

If you need to modify the state according to the `props` changes, you can use [`componentWillReceiveProps`](https://facebook.github.io/react/docs/react-component.html#componentwillreceiveprops) which is called everytime the props change.